ProcessProcedureTests.swift

```diff
@@ -8,24 +8,224 @@
 import XCTest
 import ProcedureKit
 import TestingProcedureKit
+import Foundation
 @testable import ProcedureKitMac
 
 class ProcessProcedureTests: ProcedureKitTestCase {
 
     var process: Process!
     var processProcedure: ProcessProcedure!
 
+    var launchPath: String!
+    var arguments: [String]!
+
     override func setUp() {
         super.setUp()
 
-        process = Process()
-        process.launchPath = "/bin/echo"
-        process.arguments = [ "Hello World" ]
-        processProcedure = ProcessProcedure(process: process)
+        launchPath = "/bin/echo"
+        arguments = [ "Hello World" ]
+        processProcedure = ProcessProcedure(launchPath: launchPath, arguments: arguments)
     }
 
     func test__start_process() {
         wait(for: processProcedure)
         XCTAssertProcedureFinishedWithoutErrors(processProcedure)
     }
+
+    func test__start_process_with_launchpath_only() {
+        processProcedure = ProcessProcedure(launchPath: launchPath)
+        wait(for: processProcedure)
+        XCTAssertProcedureFinishedWithoutErrors(processProcedure)
+    }
+
+    func test__cancel_process_before_launched() {
+        processProcedure.cancel()
+        wait(for: processProcedure)
+        XCTAssertProcedureCancelledWithoutErrors(processProcedure)
+    }
+
+    func test__cancel_process_after_launched() {
+        processProcedure = ProcessProcedure(launchPath: "/bin/bash/", arguments: ["-c", "sleep 2"])
+        processProcedure.addDidExecuteBlockObserver { (procedure) in
+            procedure.cancel()
+        }
+        wait(for: processProcedure, withTimeout: 1)
```

Comment:
I guess, this is okay, but I think a better way of testing this is to use the `check` API:

```swift
processProcedure = ProcessProcedure(launchPath: "/bin/bash/", arguments: ["-c", "sleep 2"])
check(procedure: processProcedure, withTimeout: 1) { procedure in
    procedure.cancel()
}
XCTAssertProcedureCancelledWithoutErrors(processProcedure)
```

Because 
a. it's more consistent throughout the test suite
b. it has less reliance on _ProcedureKit_ machinery.